packages/block-editor/src/components/block-switcher/index.js

```diff
@@ -357,8 +223,7 @@ export const BlockSwitcher = ( { clientIds } ) => {
 							<BlockSwitcherDropdownMenuContents
 								onClose={ onClose }
 								clientIds={ clientIds }
-								hasBlockStyles={ hasBlockStyles }
-								canRemove={ canRemove }
```

Comment:
`hasBlockStyles` is figured in the `useSelect` within the `BlockSwitcherDropdownMenuContents` now.  It gets sorted when the dropdown opens. `isSynced` comes from the parent BlockToolbar so it's consistent how this gets figured.